### ISSUE TITLE
i have question about dpo training process (just simple question)

### ISSUE THREAD

**Issue by YooSungHyun** [huggingface/trl#1104]
When using ppo, ref_model was frozen.
Should I also freeze ref_model when using dpo?

I checked `DPOTrainer`, and when `ref_model == None`, it was frozen after model deepcopy.

When I was training before, I accidentally entered ref_model as a variable in `DPOTrainer` and found that it trained well without `required_grad=False`. (Strange, but it trained well).

This time, I train without inputting ref_model as a variable to `DPOTrainer` (input None), and the loss does not drop.

What could be the problem?
Is it correct that dpo does not freeze ref_model?

**Comment by beardream98** [comment]
It sounds weird, but we use the ref_model to keep the output logits close to the ref_model. If we update the parameters in the ref_model, it's hard to determine whether the model has learned better, or if the ref_model's output is simply closer to the dpo model , rather than what we proposed.
